src/context.cpp

```diff
@@ -551,49 +567,48 @@ namespace librealsense
                 handle_frame_without_metadata();
         }
 
-        //Call under lock
+        // Call under lock
         void handle_match()
         {
-            T & frame = _frame_queue.front();
+            auto & fh = _frame_queue.front()._frame;
 
             json md;
             if( ! _metadata_queue.empty() )
             {
-                md = std::move( _metadata_queue.front() );
+                md = std::move( _metadata_queue.front()._md );
                 _metadata_queue.pop_front();
             }
 
             if( _on_frame_ready )
-                _on_frame_ready( std::move( frame ), std::move( md ) );
+                _on_frame_ready( std::move( fh ), std::move( md ) );
 
             _frame_queue.pop_front();
-            _prof_queue.pop_front();
         }
 
-        //Call under lock
+        // Call under lock
         void handle_frame_without_metadata()
         {
-            T & frame = _frame_queue.front();
+            auto & fh = _frame_queue.front()._frame;
 
             json md;
             if( _on_frame_ready )
-                _on_frame_ready( std::move( frame ), std::move( md ) );
+                _on_frame_ready( std::move( fh ), std::move( md ) );
 
             _frame_queue.pop_front();
-            _prof_queue.pop_front();
         }
 
         on_frame_ready _on_frame_ready = nullptr;
 
-        std::deque< T > _frame_queue;
-        std::deque< std::shared_ptr< rs2_stream_profile > > _prof_queue; // Holding shared_ptr till corresponding frame is used
-        std::deque< nlohmann::json > _metadata_queue;
+        std::deque< synced_frame > _frame_queue;
+        std::deque< synced_metadata > _metadata_queue;
         std::mutex _queues_lock;
     };
 
 
     class dds_sensor_proxy : public software_sensor
     {
+        std::map< std::string, frame_metadata_syncer > _stream_name_to_syncer;
```

Comment:
Keep it with the rest of the private members

Reply:
Done